src/RedCode/Currency/Rate/Provider/YahooCurrencyRateProvider.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace RedCode\Currency\Rate\Provider;
+
+use RedCode\Currency\ICurrency;
+use RedCode\Currency\ICurrencyManager;
+use RedCode\Currency\Rate\ICurrencyRate;
+use RedCode\Currency\Rate\ICurrencyRateManager;
+
+class YahooCurrencyRateProvider implements ICurrencyRateProvider
+{
+    const PROVIDER_NAME = 'yahoo';
+
+    public function __construct(ICurrencyRateManager $currencyRateManager, ICurrencyManager $currencyManager)
+    {
+        $this->currencyRateManager  = $currencyRateManager;
```

Comment:
These properties must be defined in the class scope.
See details of inspection [here](https://scrutinizer-ci.com/g/redco/redcode-currency-rate/inspections/2f6f9ee3-a150-4a96-890f-bb06952f21d9)